app/code/Magento/Theme/Controller/Result/JsFooterPlugin.php

```diff
@@ -39,26 +39,37 @@ public function __construct(ScopeConfigInterface $scopeConfig)
      */
     public function beforeSendResponse(Http $subject)
```

Comment:
Consider some of the variable naming suggestions/methods used here. Breaking apart into two functions makes more sense to me here also as the original function is getting way too long (consider breaking it out to even more functions). The combo of these make the code much easier to reason about.

```
    /**
     * Put all javascript to footer before sending the response.
     *
     * @param Http $subject
     * @return void
     */
    public function beforeSendResponse(Http $subject)
    {
        $content = (string)$subject->getContent();
        $isEndBodyTagFound = strpos($content, '</body') !== false;
        $shouldMoveJsToBottom = $this->scopeConfig->isSetFlag(self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM, ScopeInterface::SCOPE_STORE);

        if ($isEndBodyTagFound && $shouldMoveJsToBottom) {
            if ($scripts = $this->moveScriptsToBottom($content)) {
                $content = str_replace('</body', "$scripts\n</body", $content);
                $subject->setContent($content);
            }
        }
    }

    public function moveScriptsToBottom(&$content)
    {

        $scripts = '';
        $scriptOpen = '<script';
        $scriptClose = '</script>';
        $scriptOpenPos = strpos($content, $scriptOpen);

        while ($scriptOpenPos !== false) {
            $scriptClosePos = strpos($content, $scriptClose, $scriptOpenPos);
            $scriptPos = $scriptClosePos - $scriptOpenPos + strlen($scriptClose);
            $script = substr($content, $scriptOpenPos, $scriptPos);
            $isXMagentoTemplate = strpos($script, 'text/x-magento-template') !== false;

            if ($isXMagentoTemplate) {
                $scriptOpenPos = strpos($content, $scriptOpen, $scriptClosePos);
                continue;
            }

            $scripts .= "\n$script";
            $scriptContent = str_replace($script, '', $content);

            // Script cut out, continue search from its position.
            $scriptOpenPos = strpos($content, $scriptContent, $scriptOpenPos);
        }

        return $scripts;
    }
```